multinet/table.py

```diff
@@ -0,0 +1,109 @@
+"""Operations that deal with tables."""
+from __future__ import annotations  # noqa: T484
+from arango.collection import StandardCollection
+from arango.aql import AQL
+
+from multinet import util
+from multinet.types import EdgeTableProperties
+from multinet.errors import InternalServerError
+
+from typing import List, Set, Dict, Iterable, Union, Optional
+
+
+class Table:
+    """Table."""
+
+    def __init__(self, name: str, workspace: str, handle: StandardCollection, aql: AQL):
+        """Init Table class."""
+        self.name = name
+        self.workspace = workspace
+        self.handle = handle
+        self.aql = aql
+
+    def rows(self, offset: Optional[int] = None, limit: Optional[int] = None) -> Dict:
+        """Return the desired rows in a table."""
+        rows = self.handle.find({}, skip=offset, limit=limit)
+        count = self.handle.all().count()
+
+        return {"count": count, "rows": list(rows)}
+
+    def row(self, doc: Union[Dict, str]) -> Union[Dict, None]:
+        """Return a specific document, or `None` if not present."""
+        return self.handle.get(doc)
+
+    def row_count(self) -> int:
+        """Return the number of rows in a table."""
+        return self.handle.count()
+
+    def keys(self) -> Iterable[str]:
+        """Return all the keys in a table."""
+        return self.handle.keys()
+
+    def row_fields(self, filter_keys: bool = False) -> List[str]:
```

Comment:
`keys()` returns the keys of all the documents in the table (arangodb collection). That is, all of the `_key` fields on each document. This function returns the set of keys that are present in _each_ document. Technically speaking, it returns the keys on the first document it finds in the collection, but since we ensure a consistent schema, I think this is fine.